dxp/layout_rendering.md

```diff
@@ -0,0 +1,173 @@
+
+> All this is done leveraging the [`RequestDispatcher` API](https://docs.oracle.com/javaee/7/api/javax/servlet/RequestDispatcher.html) that can be retrieved from any [`ServletRequest`](https://javaee.github.io/javaee-spec/javadocs/javax/servlet/ServletRequest.html#getRequestDispatcher-java.lang.String-). You can see an example of how it is used in [`MainServlet.java`](https://github.com/liferay/liferay-portal/blob/2448ae35c6a8232af153acb0f366eafad59f48c4/portal-impl/src/com/liferay/portal/internal/servlet/MainServlet.java#L1119).
+
+So, whenever someone hits http://localhost:8080, the following requests are serviced by Tomcat:
+
+1. `/`: the originating external request
+2. `/web/guest`: internal request to which `/` is forwarded. Note that you can also request `/web/guest` from the browser and you get the same content.
+3. `/c/portal/layout?p_l_id=19&p_v_l_s_g_id=0`: this is internal too. The server maps the `/web/guest` request (which is a virtual path) to the configured layout, identified by a `p_l_id`. This is the request that truly generates the HTML content the user sees.
+4. `/o/js_loader_config?t=1600876854212`: this is a external request triggered by the HTML once the browser loads it. It returns the configuration of the AMD loader.
+5. `/o/classic-theme/js/main.js?browserId=firefox&minifierType=js&languageId=en_US&b=7305&t=1600791970000`: external request downloading the theme's JavaScript code.
+6. `/combo?browserId=firefox&minifierType=js&languageId=en_US&b=7305&t=1600873127973&/o/frontend-js-aui-web/aui/aui/aui.js&...&/o/frontend-js-web/liferay/workflow.js`: external request downloading DXP's core JavaScript code.
+7. `/o/js_bundle_config?t=1600876893358`: external request downloading the configuration of the legacy AUI loader.
+8. `/o/js_resolve_modules?...`: a request to resolve AMD modules. Several more of this follow as well as requests to the `combo` servlet to download the resolved modules.
+9. `/image/company_logo`: request for web site logo.
+10. `/o/classic-theme/images/clay/icons.svg`: request for Clay icon library.
+11. `/c/portal/login?p_l_id=19&windowState=exclusive`: internal request for the login portlet.
+12. `/web/guest/home-widget-?p_p_id=com_liferay_login_web_portlet_LoginPortlet&p_p_lifecycle=0&p_p_state=exclusive&p_p_mode=view&saveLastPath=false&_com_liferay_login_web_portlet_LoginPortlet_mvcRenderCommandName=%2Flogin%2Flogin`: the real path servicing the login's portlet response.
+
+In this document we are mainly interested in the first 3 which are the responsibles for rendering the HTML code you see in the browser. Specifically the third one, as the other two are simply forwards.
```

Comment:
"responsibles" isn't a word, so instead of:

> the first 3 which are the responsibles for

Do either:

> the first 3 which are responsible for

Or:

>  the first 3 which are the ones responsible for

Reply:
I would add that since we're mainly interested in the first 3 and the document is about layout rendering, that maybe we could cut the previous list/table short and just list these, or push the other 9 paths down to an appendix...

Reply:
I can also move the comment up (before the list) so that people not interested in the others skip them. Looks easier to me...

Reply:
I wrote this on top of the list:

> Note that, in this document, we are mainly interested in the first 3 which are responsible for rendering the HTML code you see in the browser (specifically the third one, as the other two are simply forwards). The rest is listed for completeness.